hoomd/md/TwoStepNPTMTK.cc

```diff
@@ -839,10 +876,90 @@ void TwoStepNPTMTK::advanceThermostat(unsigned int timestep)
     setIntegratorVariables(v);
     }
 
+void TwoStepNPTMTK::setCouple(const std::string& value)
+    {
+    bool twod = m_sysdef->getNDimensions()==2;
+    if (!(twod))
+        {
+        if (value == "none")
+            {
+            m_couple = couple_none;
+            }
+        else if (value == "xy")
+            {
+            m_couple = couple_xy;
+            }
+        else if (value == "xz")
+            {
+            m_couple = couple_xz;
+            }
+        else if (value == "yz")
+            {
+            m_couple = couple_yz;
+            }
+        else if (value == "xyz")
+            {
+            m_couple = couple_xyz;
+            }
+        }
+    // if sys_def is 2D, silently ignore any couplings that involve z
+    if (twod)
+        {
+        if (value == "none")
+            {
+            m_couple = couple_none;
+            }
+        else if (value == "xy")
+            {
+            m_couple = couple_xy;
+            }
+        else if (value == "xz")
+            {
+            m_couple = couple_none;
+            m_exec_conf->msg->warning() << "z coupling is ignored, since system is two dimensional. coupling is none." << endl;
+            }
+        else if (value == "yz")
+            {
+            m_couple = couple_none;
+            m_exec_conf->msg->warning() << "z coupling is ignored, since system is two dimensional. coupling is none." << endl;
+            }
+        else if (value == "xyz")
+            {
+            m_couple = couple_xy;
+            m_exec_conf->msg->warning() << "z coupling is ignored, since system is two dimensional. coupling is xy." << endl;
+            }
+        }
+    }
```

Comment:
I am not sure I agree with silently ignoring `z` here. I think erroring makes the most sense.

Reply:
The default value of the coupling is "xyz". Should users that run 2D systems need to opt out of this default to avoid the proposed error? What if we renamed this to "all"?